### PR TITLE
Prep release 3.0.0 rc.1

### PR DESCRIPTION
**Pull request by jan--f**

### CODE REVIEW

**Comment by bwplotka** [comment]
Should we wait for https://github.com/prometheus/prometheus/pull/15375 and https://github.com/prometheus/prometheus/pull/15372#pullrequestreview-2427780712 ?

**Comment by jan--f** [comment]
Yes will add that. Didn't know that was still coming :)

**Comment by jan--f** [comment]
Added the changelog and migration entries.

**Review by bwplotka — APPROVED** [review]
Amazing, LGTM!

**Comment by bwplotka** [comment]
Should we do prombench too?

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,7 +2,12 @@
 
 ## unreleased
 
+## 3.0.0-rc.1 / 2024-11-11
+
+* [CHANGE] Set the `GOMAXPROCS` variable automatically to match the Linux CPU quota. Use `-no-auto-gomaxprocs` to disable it. The `auto-gomaxprocs` feature flag was removed. #15376
+* [CHANGE] Set the `GOMEMLIMIT` variable automatically to match the Linux container memory limit. Use `-no-auto-gomemlimit` to disable it. The `auto-gomemlimit` feature flag was removed. #15373
```
Comment (SuperQ) [inline CHANGELOG.md:8]:
```suggestion
* [CHANGE] Set the `GOMAXPROCS` variable automatically to match the Linux CPU quota. Use `--no-auto-gomaxprocs` to disable it. The `auto-gomaxprocs` feature flag was removed. #15376
* [CHANGE] Set the `GOMEMLIMIT` variable automatically to match the Linux container memory limit. Use `--no-auto-gomemlimit` to disable it. The `auto-gomemlimit` feature flag was removed. #15373
```

**Comment by bwplotka** [comment]
We decided to include https://github.com/prometheus/prometheus/pull/15384 as well

**Review by bwplotka — APPROVED** [review]
I rebased against latest release-3.0

**Pull request merged**